spec/presenters/mfa_confirmation_presenter_spec.rb

```diff
@@ -47,4 +47,33 @@
       end
     end
   end
+
+  describe '#webauthn_platform_set_up_successful?' do
+    it 'returns false' do
+      expect(presenter.webauthn_platform_set_up_successful?).to eq(false)
+    end
+
+    context 'when f/t unlock setup is successful' do
+      let(:webauthn_platform_set_up_successful) { true }
+      let(:expected_heading) { t('titles.mfa_setup.face_touch_unlock_confirmation') }
+      let(:expected_info) { I18n.t('mfa.webauthn_platform_message') }
+      let(:presenter) do
+        described_class.new(
+          webauthn_platform_set_up_successful: webauthn_platform_set_up_successful,
+        )
+      end
+
+      it 'returns true' do
+        expect(presenter.webauthn_platform_set_up_successful?).to eq(true)
+      end
+
+      it 'shows the correct heading' do
+        expect(presenter.heading).to eq expected_heading
+      end
+
+      it 'shows the correct content' do
+        expect(presenter.info).to eq expected_info
+      end
```

Comment:
Since this is within `describe '#webauthn_platform_set_up_successful?' do`, I'd expect all of the assertions to be related to the `webauthn_platform_set_up_successful?` method. We already have `#heading?` (sic) and `#info?` (sic) `describe` blocks above, so I'd think it makes sense to move the assertions there, with their own "platform set up success" context blocks.

Reply:
So would it be:

- Test for default heading
- Test for default info
- Button
- When platform auth is successful
  - Test for heading
  - Test for info
 - Test for `#show_skip_additional_mfa_link?` functionality
 - Test for `#webauthn_platform_setup_successful?` functionality

Reply:
I don't know that there's any hard rules around it, but I generally prefer `describe` blocks for each method to be at the top-level, and then relevant context within.

So roughly:

```rb
RSpec.describe MfaConfirmationPresenter do
  describe '#heading' do
    context 'with successful webauthn platform setup' do
      it '...'
    end

    context 'without successful webauthn platform setup' do
      it '...'
    end
  end

  describe '#info' do
    context 'with successful webauthn platform setup' do
      it '...'
    end

    context 'without successful webauthn platform setup' do
      it '...'
    end
  end

  describe '#button' do
    it '...'
  end

  describe '#show_skip_additional_mfa_link?' do
    it '...'
  end
end
```

I'd suggest we could make `webauthn_platform_set_up_successful?` a private method and avoid writing tests for it if the only place we're using it is within `#heading` and `#info` (which would already have test coverage via its `context` branches).

Reply:
Ah, I see. My thinking was to have everything set up in one `describe 'when platform authenticator is set'` block to cover all logic that is related to the behavior, but I can see how that setup also makes sense.